resources/destination.rb

```diff
@@ -16,40 +16,84 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-property :config_dir, String, default: '/etc/syslog-ng/destination.d'
-property :cookbook, String
-property :source, String
-property :driver, [String, Array]
-property :path, [String, Array]
-property :parameters, [Hash, Array]
-property :configuration, Array
-property :description, String
-property :multiline, [true, false], default: false
+include SyslogNg::Cookbook::GeneralHelpers
 
-action :create do
-  extend SyslogNg::CommonHelpers
+property :config_dir, String,
+          default: lazy { "#{syslog_ng_config_dir}/destination.d" },
+          description: 'Directory to create configuration file in'
 
-  destination = parameter_builder(driver: new_resource.driver, path: new_resource.path, parameters: new_resource.parameters, configuration: new_resource.configuration).each do |config|
-    raise "destination: Expected driver configuration to be a Hash, got #{config.class}" unless config.is_a?(Hash)
-    config.each do |_, b|
-      b.compact!
-    end
-  end
+property :cookbook, String,
+          default: 'syslog_ng',
+          description: 'Cookbook to source configuration file template from'
+
+property :template, String,
+          default: 'syslog-ng/destination.conf.erb',
+          description: 'Template to use to generate the configuration file'
+
+property :owner, String,
+          default: lazy { syslog_ng_user },
```

Comment:
```suggestion
          default: lazy { default_syslog_ng_user },
```

I would prefix these defaults as such. It's ended up as a bit of a pattern from other cookbooks, and it makes the intention clearer